stdlib/test/test_utils/types.mojo

```diff
@@ -108,14 +108,20 @@ struct MoveCounter[T: CollectionElementNew](
         self.move_count = existing.move_count
 
 
+# TODO: Pass directly a UnsafePointer when possible, instead of an Int.
+# Otherwise we get "argument #2 cannot be converted from 'UnsafePointer[List[Int], 0, 0]' to 'UnsafePointer[List[Int], 0, 0]'"
+# This bug also appears if we pass the list as borrow and grab the address in the constructor, in
+# which case we get "argument #2 cannot be converted from 'List[Int]' to 'List[Int]'"
```

Comment:
**Question** Does this work if you use `rebind[UnsafePointer[List[Int]]](ptr)` wherever the pointer is passed? That might be a simpler workaround (if it does work).

Reply:
I'm not sure I understand where the rebind should go, but I tried this:
```mojo
@value
struct ValueDestructorRecorder(CollectionElement):
    var value: Int
    var destructor_counter: UnsafePointer[List[Int]]

    fn __init__(inout self, value: Int, destructor_counter: UnsafePointer[List[Int]]):
        self.value = value
        self.destructor_counter = rebind[UnsafePointer[List[Int]]](destructor_counter)

    fn __init__(inout self, *, other: Self):
        self.value = other.value
        self.destructor_counter = other.destructor_counter

    fn __del__(owned self):
        self.destructor_counter[].append(self.value)
```
And I get the following error:

```
/projects/open_source/mojo/stdlib/test/memory/test_maybe_uninitialized.mojo:26:32: error: no matching function in initialization
        ValueDestructorRecorder(
        ~~~~~~~~~~~~~~~~~~~~~~~^
/projects/open_source/mojo/stdlib/test/test_utils/types.mojo:120:8: note: candidate not viable: argument #2 cannot be converted from 'UnsafePointer[List[Int], 0, 0]' to 'UnsafePointer[List[Int], 0, 0]'
    fn __init__(inout self, value: Int, destructor_counter: UnsafePointer[List[Int]]):
       ^
/projects/open_source/mojo/stdlib/test/test_utils/types.mojo:124:8: note: candidate not viable: missing 1 required keyword-only argument: 'other'
    fn __init__(inout self, *, other: Self):
       ^
mojo: error: failed to parse the provided Mojo source module
```